Add version synchronizer (#685)
kyma-project/test-infra: development/tools/cmd/synchronizer/README.md

# Component synchronizer

## Overview

This command walks recursively through all Kyma repository components and searches makefile command which returns localization of component version. 

> Next compares component version with actual component git hash commit. If component is out of date then an appropriate warning will be send to slack channel.

Author: 👍 I added more general information but I also mentioned about slack.

## Usage

> The command is run periodically by prow job. To run script localy set below environment variable and run as regular go application

Author: I removed part about prowJobs.

# Component synchronizer

## Overview

> This command walks recursively through all Kyma repository components and searches makefile command which returns localization of component version. 

Author: :+1:

Next compares component version with actual component git hash commit. If component is out of date then an appropriate warning will be displayed in logs.
It is also possible to send alert message to other applications (currently available application is slack). 

## Usage

> To run script localy set below environment variable and run as regular go application

Author: :+1:

> To run script localy set below environment variables and run as regular go application

Reviewer: ```suggestion
To run the script locally, set the [environment variables](#environment-variables), navigate to `/development/tools`, and ensure that all vendor dependencies are up to date. Run:
```

ensuring earlier all vendor dependencies are up to date.
Run from `/development/tools`
```
dep ensure -v -vendor-only
```

> then run from current directory

Reviewer: ```suggestion
Then, run this command:
```

```
go run main.go
```

### Environment variables

| Name                                  | Required  | Description                              |
| :------------------------------------ | :------:  | :--------------------------------------- |

> | **KYMA_PROJECT_DIR**                  |    Yes    | Path to the `kyma-project` directory which contains the Kyma source code. |

Author: `kyma-project/kyma`, fixed

> | **KYMA_PROJECT_DIR**                  |    Yes    | Path to the `kyma-project/kyma` directory which contains the Kyma source code. |

Reviewer: ```suggestion
| **KYMA_PROJECT_DIR**                  |    YES    | A path to the `kyma-project/kyma` directory which contains the Kyma source code. |
```

> | **SLACK_CLIENT_TOKEN**                |    No     | Token to slack where message about out of date components will be sent. If not set message will be not set, all information will be available in logs. |

Reviewer: ```suggestion
| **SLACK_CLIENT_TOKEN**                |    NO     | A token to the Slack channel where messages about outdated components are sent. If not set, the messages are not sent and the information about the outdated components is available in logs. |
```

> | **STABILITY_SLACK_CLIENT_CHANNEL_ID** |    No     | Channel name where message about out of date components will be sent. If not set message will be not set, all information will be available in logs. |

Reviewer: ```suggestion
| **STABILITY_SLACK_CLIENT_CHANNEL_ID** |    NO     | An ID of the Slack channel where messages about outdated components are sent. If not set, the messages are not sent and the information about the outdated components is available in logs. |
```

> | **OUT_OF_DATE_DAYS**                  |    No     | Number of days after which component will be treated as out of date (default value is 3). |

Reviewer: ```suggestion
| **OUT_OF_DATE_DAYS**                  |    NO     | A number of days after which a component is treated as outdated. The default value is `3`. |
```
